### DISCUSSION TITLE
[Nuxt3] Accessing plugins from server api

### DISCUSSION THREAD

**Discussion by DNA** [nuxt/nuxt#17096] [Questions]
Hi guys! 

I'm tinkering with Nuxt3 and I'm trying to add a db connection to a server api:

I've built a plugin to make a Knex instance accessible:
```typescript
// plugins/database.server.ts
import knex from 'knex';

const config = useRuntimeConfig()

const database = knex({
  client: 'mysql',
  connection: config.database
});

export default defineNuxtPlugin(() => {
  return {
    provide: { database }
  }
})
```

And used it on my api:
```typescript
// server/api/services.get.ts

export default defineEventHandler(async () => {
  const Services = useNuxtApp().$database('Services')

  return Services.select('*')
})
```

vscode doesn't give me any warning or error, but I got an error trying to access the `/api/service` endpoint:

```
[nuxt] [request error] useNuxtApp is not defined
  at createError (./node_modules/h3/dist/index.mjs:191:15)
  at Server.nodeHandler (./node_modules/h3/dist/index.mjs:381:21)
  at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Is this the best approach for what I'm trying to do here? And if so, what I'm doing wrong that I can't use `userNuxtApp` here?

Thanks! :D

**Comment by Flanker72** [comment]
You may check [this article](https://medium.com/@flanker72/nuxt3-complex-solutions-database-integration-8df941f0fb82) about how to connect to a DB in a proper way.

**Comment by DNA** [comment]
Thanks for the article, @Flanker72! 😄 

But while it greatly helps with the database aspect used on the example, it doesn't explain the main question: _Why can't I access plugins from a server api?_ 

Are they not meant to be used this way?
Is it only for SSR context?
Is it a design decision? 
Or... Have I got everything wrong? 😅

**Reply by Flanker72** [reply]
As I can understand from docs and code, plugins are used for client-side part of the application, they are not suitable for server usage. For server-side plugins you can use Nitro plugins instead (as described in the given article). I was blocked by the same problem during database connection, so that article was written after some investigation in this way) Hope, it can help you to achieve the result.
